sdk/monitor/azure-monitor-query/azure/monitor/query/_log_query_client.py

```diff
@@ -0,0 +1,146 @@
+
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential
+
+
+class LogsClient(object):
+    """LogsClient
+
+    :param credential: The credential to authenticate the client
+    :type credential: ~azure.core.credentials.TokenCredential
+    :keyword endpoint: The endpoint to connect to. Defaults to 'https://api.loganalytics.io/v1'.
+    :paramtype endpoint: str
+    """
+
+    def __init__(self, credential, **kwargs):
+        # type: (TokenCredential, Any) -> None
+
+        self._endpoint = kwargs.pop('endpoint', 'https://api.loganalytics.io/v1')
+        self._client = MonitorQueryClient(
+            credential=credential,
+            authentication_policy=get_authentication_policy(credential),
+            base_url=self._endpoint,
+            **kwargs
+        )
+        self._query_op = self._client.query
+
+    def query(self, workspace_id, query, **kwargs):
+        # type: (str, str, Any) -> LogsQueryResults
+        """Execute an Analytics query.
+
+        Executes an Analytics query for data.
+
+        :param workspace_id: ID of the workspace. This is Workspace ID from the Properties blade in the
+         Azure portal.
+        :type workspace_id: str
+        :param query: The Analytics query. Learn more about the `Analytics query syntax
+         <https://azure.microsoft.com/documentation/articles/app-insights-analytics-reference/>`_.
+        :type query: str
+        :keyword str timespan: Optional. The timespan over which to query data. This is an ISO8601 time
+         period value.  This timespan is applied in addition to any that are specified in the query
+         expression.
+        :keyword int server_timeout: the server timeout in seconds. The default timeout is 3 minutes,
+         and the maximum timeout is 10 minutes.
+        :keyword bool include_statistics: To get information about query statistics.
+        :keyword bool include_render: In the query language, it is possible to specify different render options.
+         By default, the API does not return information regarding the type of visualization to show.
+         If your client requires this information, specify the preference
+        :keyword workspaces: A list of workspaces that are included in the query.
+        :paramtype workspaces: list[str]
+        :keyword qualified_names: A list of qualified workspace names that are included in the query.
+        :paramtype qualified_names: list[str]
+        :keyword workspace_ids: A list of workspace IDs that are included in the query.
+        :paramtype workspace_ids: list[str]
+        :keyword azure_resource_ids: A list of Azure resource IDs that are included in the query.
+        :paramtype azure_resource_ids: list[str]
+        :return: QueryResults, or the result of cls(response)
+        :rtype: ~azure.monitor.query.LogsQueryResults
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        timespan = kwargs.pop("timespan", None)
```

Comment:
Is it easy for the user to construct the ISO 8601 formatted time intervals? Is it better to provide a simpler way to specify the start and end (datetime/duration)?

```
Start and end, such as "2007-03-01T13:00:00Z/2008-05-11T15:30:00Z"
Start and duration, such as "2007-03-01T13:00:00Z/P1Y2M10DT2H30M"
Duration and end, such as "P1Y2M10DT2H30M/2008-05-11T15:30:00Z"
Duration only, such as "P1Y2M10DT2H30M", with additional context information
```

Reply:
removed timespan kwarg to replace it with start_time, end_time and duration - we'll magically construct the timespan for the users now